---
title: "Validation"
author: "Helen K."
date: "11/1/2021"
output: pdf_document
---

# Dataset setup

```{r setup, echo=F, message=FALSE, warning=FALSE}

# Load Rdata produced from script Basic_data_creation_for_analysis.R
load("./Basic_data_for_analysis.RData") 
knitr::opts_chunk$set(echo = TRUE)
library("dplyr")
library("ggplot2")
library("biomaRt")
library("patchwork")
library("gprofiler2")
library("DESeq2")
library("tidyr")
library("tibble")
library("PCAtools")
library("ggtext")


## Create a smaller sample info variable for easier plotting later.
clone_condition <- factor(clone_condition, levels=c("TBJ360_DN", "TBJ360_Blue", "TBJ360_DP", "TBJ360_Red", "TBX4B_DN", "TBX4B_Blue", "TBX4B_DP", "TBX4B_Red"))
                          
sample_info$sample <-rownames(sample_info)
sample_info$clone_condition <- clone_condition







# save.image(file = "./Data_and_metadata.RData")
```

# 3.60 2 clusters

## Scaled data with Kmeans clustering using "MacQueen" {.tabset .tabset-pills}


```{r echo=TRUE, fig.height=7, fig.width=7}
vst_TBJ_dds<-varianceStabilizingTransformation(TBJ_dds_rsem, blind=F)
vst_TBX_dds<-varianceStabilizingTransformation(TBX_dds_rsem, blind=F)

# Subset DE genes identified by LRT from transformed matrix.
vst_tbj_regulated <- assay(vst_TBJ_dds)[dds_res_sig_TBJ,]

# Transfer data to a new clustering input variable.
tbj_clustering_input <- vst_tbj_regulated

# Re-order column names
tbj_clustering_input <-tbj_clustering_input[,c("TBJ360_DN1", "TBJ360_DN2", "TBJ360_Blue1", "TBJ360_Blue2", "TBJ360_DP1", "TBJ360_DP2", "TBJ360_Red1", "TBJ360_Red2")]

# Name by symbol.
rownames(tbj_clustering_input) <- gene_names_rsem$hgnc_symbol[match(rownames(tbj_clustering_input), gene_names_rsem$ensembl_gene_id)] 

# Only include unique.
tbj_clustering_input<-tbj_clustering_input[!duplicated(row.names(tbj_clustering_input)),]

# Z-score the data.
scaled_counts <- tbj_clustering_input%>%
  t()%>%
  scale(scale=T)%>%
  t()

# Define nr of k
k<-2
set.seed(1234)

kmeans_clust <- kmeans(scaled_counts, centers=k, iter.max = 10000, nstart = 100, algorithm="MacQueen")

# Extract cluster info.
clusters <- kmeans_clust$cluster

# Turn into a tibble for better access.
TBJ_genes_in_clusters <- clusters %>% 
  # turn the named vector into a tibble
  tibble::enframe() %>% 
  # rename some of the columns
  dplyr::rename(gene = name, cluster = value) %>%  
  arrange(cluster)

```


### Enrichment analysis of 3.60 clusters.

```{r echo=TRUE, fig.height=20, fig.width=22}


# Set the correct ensembl version
set_base_url("https://biit.cs.ut.ee/gprofiler_archive3/e100_eg47_p14")
# Load the hallmarks gmt
token_all_mgsig = upload_GMT_file(gmtfile = "./h.all.v7.2.symbols.gmt")

# Extract background genes. 
TBJ_all_DE_bg_symbols <- dds_res_TBJ[(complete.cases(dds_res_TBJ)),]
TBJ_all_DE_bg_symbols<-row.names(TBJ_all_DE_bg_symbols)
TBJ_all_DE_bg_symbols <- gene_names_rsem$hgnc_symbol[match(TBJ_all_DE_bg_symbols, gene_names_rsem$ensembl_gene_id)]

cluster1_tbj <- TBJ_genes_in_clusters$gene[TBJ_genes_in_clusters$cluster ==1]

# Enrichment analysis using gprofiler

cluster1_tbj_gprofiler <- gost(list(regulated=cluster1_tbj),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom_annotated",
                               custom_bg = as.character(TBJ_all_DE_bg_symbols))



cluster2_tbj <- TBJ_genes_in_clusters$gene[TBJ_genes_in_clusters$cluster ==2]

cluster2_tbj_gprofiler <- gost(list(regulated=cluster2_tbj),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom_annotated",
                               custom_bg = as.character(TBJ_all_DE_bg_symbols))



# write.table(TBJ_all_DE_bg_symbols, file="./TBJ_cluster/2_cluster/TBJ_all_DE_bg_symbols.txt", quote = F, row.names = F, col.names = F)
```

# TBX4B 2 clusters

## Scaled data with Kmeans clustering using "MacQueen" {.tabset .tabset-pills}


```{r echo=F, fig.height=7, fig.width=7, message=FALSE, warning=FALSE}

# Steps as above for clone 3.60

vst_tbx_regulated <- assay(vst_TBX_dds)[dds_res_TBX_sig,]
tbx_clustering_input <- vst_tbx_regulated
tbx_clustering_input <-tbx_clustering_input[,c("TBX4B_DN1", "TBX4B_DN2", "TBX4B_Blue1", "TBX4B_Blue2", "TBX4B_DP1", "TBX4B_DP2", "TBX4B_Red1", "TBX4B_Red2")]

rownames(tbx_clustering_input) <- gene_names_rsem$hgnc_symbol[match(rownames(tbx_clustering_input), gene_names_rsem$ensembl_gene_id)] 

tbx_clustering_input <-tbx_clustering_input[!duplicated(row.names(tbx_clustering_input)),]

scaled_counts_tbx <- tbx_clustering_input%>%
  t()%>%
  scale(scale=T)%>%
  t()

set.seed(1234)
k=2

# Re-using variable names for intermediate results.

kmeans_clust <- kmeans(scaled_counts_tbx, centers=k, iter.max = 10000, nstart = 100, algorithm="MacQueen")
clusters <- kmeans_clust$cluster


TBX_genes_in_clusters <- clusters %>% 
  tibble::enframe() %>% 
  dplyr::rename(gene = name, cluster = value)

# In case clustering has changed the order of the clusters (cluster 1 should include 2725 up-regulated genes including HTLV-1 plus and cluster 2 2073 down-regulated genes) use the following code. Important for correct plotting in manuscript_plots.Rmd code
TBX_genes_in_clusters<-TBX_genes_in_clusters %>%
  mutate(cluster=case_when(
    cluster == "1" ~ "2",
    cluster == "2" ~ "1",
  ))

```


```{r fig.height=20, fig.width=20, message=FALSE, warning=FALSE, include=FALSE}

TBX_all_DE_bg_symbols <- dds_res_TBX[(complete.cases(dds_res_TBX)),]
TBX_all_DE_bg_symbols<-row.names(TBX_all_DE_bg_symbols)
TBX_all_DE_bg_symbols <- gene_names_rsem$hgnc_symbol[match(TBX_all_DE_bg_symbols, gene_names_rsem$ensembl_gene_id)]

cluster1_tbx <- TBX_genes_in_clusters$gene[TBX_genes_in_clusters$cluster ==1]

cluster1_tbx_gprofiler <- gost(list(regulated=cluster1_tbx),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom",
                               custom_bg = as.character(TBX_all_DE_bg_symbols))



cluster2_tbx <- TBX_genes_in_clusters$gene[TBX_genes_in_clusters$cluster ==2]

cluster2_tbx_gprofiler <- gost(list(regulated=cluster2_tbx),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom",
                               custom_bg = as.character(TBX_all_DE_bg_symbols))





# write.table(TBX_all_DE_bg_symbols, file="./TBX_cluster/cluster_2_TBX/TBX_all_DE_bg_symbols.txt", quote = F, row.names = F, col.names = F)
```


# Clustering differentially expressed genes in both clones - overlap.

```{r echo=F, fig.height=6, fig.width=8}

# Obtain all the overlapping DE genes
union_names <- TBX_genes_in_clusters[TBX_genes_in_clusters$gene %in% TBJ_genes_in_clusters$gene,]

# Order DESeq2 results based on pvalue column - this is for plotting.
TBJ_p_values<- as.data.frame(dds_res_TBJ)%>%
  subset(padj < 0.01)%>%
  arrange(desc(baseMean))%>%
  arrange((pvalue))%>%
  dplyr::select(pvalue)

# Rank these from 1-END
TBJ_p_values$pvalue<-1:(length(TBJ_p_values$pvalue))

TBX_p_values<- as.data.frame(dds_res_TBX)%>%
  subset(padj < 0.01)%>%
  arrange(desc(baseMean))%>%
  arrange((pvalue))%>%
  dplyr::select(pvalue)
TBX_p_values$pvalue<-1:(length(TBX_p_values$pvalue))


# Change names to HGNC symbols.
TBJ_p_values<-as.matrix(TBJ_p_values)
rownames(TBJ_p_values) <- gene_names_rsem$hgnc_symbol[match(rownames(TBJ_p_values),gene_names_rsem$ensembl_gene_id)]
TBX_p_values<-as.matrix(TBX_p_values)
rownames(TBX_p_values) <- gene_names_rsem$hgnc_symbol[match(rownames(TBX_p_values),gene_names_rsem$ensembl_gene_id)]

# Add the rank information
union_names$tbx_cluster <- TBX_p_values[match(union_names$gene, rownames(TBX_p_values)),]
union_names$tbj_cluster <- TBJ_p_values[match(union_names$gene, rownames(TBJ_p_values)),]
union_names<-union_names[c(1, 3, 4)]

# Take the mean rank
union_names$average_mean_p_value_rank <- (union_names$tbx_cluster + union_names$tbj_cluster)/2
union_names<-union_names%>%
  arrange(average_mean_p_value_rank)


# Subset the VST matrices for each clone for the corresponding vst values for these genes
overlap_in_tbx_vst<- scaled_counts_tbx[match(union_names$gene, row.names(scaled_counts_tbx)),]
overlap_in_tbx_vst<-overlap_in_tbx_vst[unique(rownames(overlap_in_tbx_vst)),]
overlap_in_tbj_vst<- scaled_counts[match(union_names$gene, row.names(scaled_counts)),]
overlap_in_tbj_vst<-overlap_in_tbj_vst[unique(rownames(overlap_in_tbj_vst)),]




# Merge these into one matrix
one_large_overlap_matrix <- merge(overlap_in_tbx_vst, overlap_in_tbj_vst, by=0, sort=F)
rownames(one_large_overlap_matrix)<-rownames(overlap_in_tbx_vst)
one_large_overlap_matrix<-one_large_overlap_matrix[,-1]


k=5
set.seed(1234)
kmeans_clust <- kmeans(one_large_overlap_matrix, centers=k, iter.max = 10000, nstart = 100, algorithm="MacQueen")
clusters <- kmeans_clust$cluster

Both_cluster_genes <- clusters %>%
  tibble::enframe() %>%
  dplyr::rename(gene = name, cluster = value)

# Obtain gene numbers in each cluster for plot formatting.
for(i in 1:k){
  Both_cluster_genes$cluster[(Both_cluster_genes$cluster == i)] <- paste0(Both_cluster_genes$cluster[(Both_cluster_genes$cluster == i)],": ", sum(Both_cluster_genes$cluster==i))
}
sample_info$clone_condition<-factor(sample_info$clone_condition, levels=unique(sample_info$clone_condition))


trans_cts_long<-as.data.frame(one_large_overlap_matrix) %>%
  tibble::rownames_to_column(var="gene")%>%
  pivot_longer(cols=c(2:17), 
               names_to = "sample", 
               values_to = "cts")%>%
  full_join(sample_info, by=("sample"))



trans_cts_mean <- trans_cts_long %>% 
  group_by(gene) %>%
  mutate(cts_scaled=(cts-mean(cts))/sd(cts))%>%
  group_by(gene, clones, condition) %>% 
  summarise(mean_cts_scaled = mean(cts_scaled)) %>% 
  ungroup()


# Make sure factor levels are accurate and re-name them.
trans_cts_mean$clones <- relevel((trans_cts_mean$clones), "TBJ360")
levels(trans_cts_mean$clones)<- c("3.60" ,"TBX4B")

trans_cts_cluster2_both <- trans_cts_mean %>% 
  inner_join(Both_cluster_genes, by = "gene")

# Re-order the clusters for plotting making sure the correct clusters are renamed
trans_cts_cluster2_both<-trans_cts_cluster2_both %>%
  mutate(cluster=case_when(
    clones=="TBX4B" & cluster == "2: 279" ~ "5: 279",
    clones=="TBX4B" & cluster == "1: 1299" ~ "3: 1299",
    clones=="TBX4B" & cluster == "3: 1093" ~ "1: 1093",
    clones=="TBX4B" & cluster == "4: 882" ~ "2: 882",
    clones=="TBX4B" & cluster == "5: 298" ~ "4: 298",
    clones=="3.60" & cluster == "2: 279" ~ "5: 279",
    clones=="3.60" & cluster == "1: 1299" ~ "3: 1299",
    clones=="3.60" & cluster == "3: 1093" ~ "1: 1093",
    clones=="3.60" & cluster == "4: 882" ~ "2: 882",
    clones=="3.60" & cluster == "5: 298" ~ "4: 298"
  ))


trans_cts_cluster2_both<-trans_cts_cluster2_both %>%
  mutate(cluster_col=case_when(
    clones=="TBX4B" & cluster == "5: 279" ~ "Down",
    clones=="TBX4B" & cluster == "1: 1093" ~ "Up",
    clones=="TBX4B" & cluster == "2: 882" ~ "Up",
    clones=="TBX4B" & cluster == "4: 298" ~ "Up",
    clones=="TBX4B" & cluster == "3: 1299" ~ "Down",
    clones=="3.60" & cluster == "5: 279" ~ "Up",
    clones=="3.60" & cluster == "1: 1093" ~ "Up",
    clones=="3.60" & cluster == "2: 882" ~ "Up",
    clones=="3.60" & cluster == "4: 298" ~ "Down",
    clones=="3.60" & cluster == "3: 1299" ~ "Down"
  ))

Both_cluster_genes <- clusters %>%
  tibble::enframe() %>%
  dplyr::rename(gene = name, cluster = value)

```




```{r fig.height=10, fig.width=15, include=FALSE}

# Join background information for enrichment analysis
BG_both<-c(TBX_all_DE_bg_symbols, TBJ_all_DE_bg_symbols)
BG_both<-unique(BG_both)

# Obtain genes for each cluster
cluster1_both <- unique(trans_cts_cluster2_both$gene[trans_cts_cluster2_both$cluster =="1: 1093"])

cluster2_both <- unique(trans_cts_cluster2_both$gene[trans_cts_cluster2_both$cluster =="2: 882"])

cluster3_both <- unique(trans_cts_cluster2_both$gene[trans_cts_cluster2_both$cluster =="3: 1299"])

cluster4_both <- unique(trans_cts_cluster2_both$gene[trans_cts_cluster2_both$cluster =="4: 298"])

cluster5_both <- unique(trans_cts_cluster2_both$gene[trans_cts_cluster2_both$cluster =="5: 279"])


# write.table(cluster4_both, file="./BOTH_clusters/3_60_TBX4B_overlap_cluster4_genes.txt", quote = F, row.names = F, col.names = F)
```

## Hallmarks results {.tabset .tabset-pills}

```{r fig.height=10, fig.width=15, include=FALSE}


# Set the correct ensembl version
set_base_url("https://biit.cs.ut.ee/gprofiler_archive3/e100_eg47_p14")

# Upload gmt. If it gives error 400, run the upload again - probably something happened to the token
token_all_mgsig = upload_GMT_file(gmtfile = "./h.all.v7.2.symbols.gmt")


cluster1_both_gprofiler <- gost(list(regulated=cluster1_both),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom",
                               custom_bg = as.character(BG_both))



cluster2_both_gprofiler <- gost(list(regulated=cluster2_both),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom",
                               custom_bg = as.character(BG_both))

cluster3_both_gprofiler <- gost(list(regulated=cluster3_both),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom",
                               custom_bg = as.character(BG_both))

cluster4_both_gprofiler <- gost(list(regulated=cluster4_both),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom",
                               custom_bg = as.character(BG_both))

cluster5_both_gprofiler <- gost(list(regulated=cluster5_both),
                               organism=token_all_mgsig,
                               ordered_query = F,
                               multi_query = F,
                               significant = T,
                               evcodes=T,
                               exclude_iea = T,
                               correction_method = "fdr",
                               domain_scope = "custom",
                               custom_bg = as.character(BG_both))

# write.table(cluster1_tbx, file="./TBX_cluster/cluster_4_TBX/cluster1_tbx.txt", quote = F, row.names = F, col.names = F)
```

# Save data

```{r echo=F}
save.image(file = "./Validation_clustering.RData")

```

sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] remotes_2.4.2               extrafont_0.17              ggtext_0.1.1                PCAtools_2.6.0             
 [5] ggrepel_0.9.1               tibble_3.1.6                tidyr_1.2.0                 gprofiler2_0.2.1           
 [9] patchwork_1.1.1             ggplot2_3.3.5               ashr_2.2-54                 dplyr_1.0.8                
[13] magrittr_2.0.3              tximport_1.22.0             BiocManager_1.30.16         DESeq2_1.34.0              
[17] SummarizedExperiment_1.24.0 Biobase_2.54.0              MatrixGenerics_1.6.0        matrixStats_0.61.0         
[21] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1         IRanges_2.28.0              S4Vectors_0.32.4           
[25] BiocGenerics_0.40.0         biomaRt_2.50.3             

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3          ellipsis_0.3.2            markdown_1.1              XVector_0.34.0           
  [5] gridtext_0.1.4            farver_2.1.0              bit64_4.0.5               AnnotationDbi_1.56.2     
  [9] fansi_1.0.3               xml2_1.3.3                splines_4.1.2             sparseMatrixStats_1.6.0  
 [13] cachem_1.0.6              geneplotter_1.72.0        knitr_1.38                jsonlite_1.8.0           
 [17] Rttf2pt1_1.3.10           annotate_1.72.0           dbplyr_2.1.1              png_0.1-7                
 [21] compiler_4.1.2            httr_1.4.2                dqrng_0.3.0               lazyeval_0.2.2           
 [25] assertthat_0.2.1          Matrix_1.4-1              fastmap_1.1.0             cli_3.2.0                
 [29] BiocSingular_1.10.0       htmltools_0.5.2           prettyunits_1.1.1         tools_4.1.2              
 [33] rsvd_1.0.5                gtable_0.3.0              glue_1.6.2                GenomeInfoDbData_1.2.7   
 [37] reshape2_1.4.4            rappdirs_0.3.3            Rcpp_1.0.8.3              vctrs_0.4.0              
 [41] Biostrings_2.62.0         extrafontdb_1.0           DelayedMatrixStats_1.16.0 xfun_0.30                
 [45] stringr_1.4.0             beachmat_2.10.0           lifecycle_1.0.1           irlba_2.3.5              
 [49] XML_3.99-0.9              zlibbioc_1.40.0           scales_1.1.1              hms_1.1.1                
 [53] parallel_4.1.2            RColorBrewer_1.1-3        yaml_2.3.5                curl_4.3.2               
 [57] memoise_2.0.1             yulab.utils_0.0.4         stringi_1.7.6             RSQLite_2.2.12           
 [61] SQUAREM_2021.1            genefilter_1.76.0         ScaledMatrix_1.2.0        filelock_1.0.2           
 [65] BiocParallel_1.28.3       truncnorm_1.0-8           rlang_1.0.2               pkgconfig_2.0.3          
 [69] bitops_1.0-7              evaluate_0.15             lattice_0.20-45           invgamma_1.1             
 [73] purrr_0.3.4               labeling_0.4.2            htmlwidgets_1.5.4         cowplot_1.1.1            
 [77] bit_4.0.4                 tidyselect_1.1.2          plyr_1.8.7                R6_2.5.1                 
 [81] generics_0.1.2            DelayedArray_0.20.0       DBI_1.1.2                 pillar_1.7.0             
 [85] withr_2.5.0               survival_3.3-1            KEGGREST_1.34.0           RCurl_1.98-1.6           
 [89] mixsqp_0.3-43             crayon_1.5.1              utf8_1.2.2                BiocFileCache_2.2.1      
 [93] plotly_4.10.0             rmarkdown_2.13            progress_1.2.2            locfit_1.5-9.5           
 [97] grid_4.1.2                data.table_1.14.2         blob_1.2.3                digest_0.6.29            
[101] xtable_1.8-4              gridGraphics_0.5-1        munsell_0.5.0             viridisLite_0.4.0        
[105] ggplotify_0.1.0   